packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -345,8 +345,17 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
       return;
     }
 
-    const principal = this.parsePermissionPrincipal(permission.principal);
-    const { sourceAccount, sourceArn } = this.parseConditions(permission.principal) ?? {};
+    let principal = this.parsePermissionPrincipal(permission.principal);
+
+    let { sourceArn, sourceAccount, principalOrgID } = this.validateConditionCombinations(permission.principal) ?? {};
+
+    // Lambda does not actually accept an organization principal here.
+    // so the principal becomes '*' and the organization ID will be sent to
+    // the 'principalOrgID' property.
+    if (permission.principal.toString().startsWith('OrganizationPrincipal')) {
```

Comment:
Can we not do without this type test? Can we not parse the policyfragment conditions and detect the principal from there?

Reply:
Or better yet, doesn't `validateCombinations` return this value already?

Reply:
You're right. It's redundant, but I had to fix a bug in `isPrincipalWithConditions` to make it happen.